tools/download-srs.sh

```diff
@@ -0,0 +1,225 @@
+    if [ "$size_bytes" -gt "$g1_total_size" ]; then
+        echo "Error: Requested g1 size ($size_bytes bytes) is larger than the source g1.point file ($g1_total_size bytes)"
+        return 1
+    fi
+    
+    if [ "$g2_size_bytes" -gt "$g2_total_size" ]; then
+        echo "Error: Requested g2 size ($g2_size_bytes bytes) is larger than the source g2.point file ($g2_total_size bytes)"
+        return 1
+    fi
+    
+    # Calculate the start position for g2.trailing.point
+    # Since we want the last G2_SIZE_BYTES of the file
+    local g2_trailing_start=$((g2_total_size - g2_size_bytes))
+    
+    # g2_trailing_start should never be negative due to the validation above,
+    # but if it is somehow negative, we should fail rather than download incorrect data
+    if [ "$g2_trailing_start" -lt 0 ]; then
+        echo "Error: Cannot download g2.trailing.point of size ${g2_size_bytes} bytes - insufficient source data"
+        return 1
+    fi
+    
+    # Download g1.point file
+    if ! download_file "${base_url}/g1.point" "${output_dir}/g1.point" "0-$((size_bytes - 1))" "g1.point (${size_bytes} bytes)"; then
+        return 1
+    fi
+    
+    # Download g2.point file
+    if ! download_file "${base_url}/g2.point" "${output_dir}/g2.point" "0-$((g2_size_bytes - 1))" "g2.point (${g2_size_bytes} bytes)"; then
+        return 1
+    fi
+    
+    # Download g2.trailing.point file
+    if ! download_file "${base_url}/g2.point" "${output_dir}/g2.trailing.point" "${g2_trailing_start}-$((g2_total_size - 1))" "g2.trailing.point (${g2_size_bytes} bytes from the end of g2.point)"; then
+        return 1
+    fi
+    
+    # Generate hashes for verification
+    echo "Generating verification hashes..."
+    
+    # Check if all files exist before generating hashes
+    for file in "${output_dir}/g1.point" "${output_dir}/g2.point" "${output_dir}/g2.trailing.point"; do
+        if [ ! -f "$file" ]; then
+            echo "Error: File $file not found. Cannot generate hashes."
+            return 1
+        fi
+    done
+    
+    # Generate the hash file
+    {
+        echo "# SRS files hashes for blob size ${size_bytes} bytes"
+        echo "# Generated on $(date -u "+%Y-%m-%d %H:%M:%S UTC")"
+        echo "# Format: SHA256 (filename)"
+        echo ""
+        sha256sum "${output_dir}/g1.point" | sed "s|${output_dir}/||"
+        sha256sum "${output_dir}/g2.point" | sed "s|${output_dir}/||"
+        sha256sum "${output_dir}/g2.trailing.point" | sed "s|${output_dir}/||"
+    } > "${hash_file}"
+    
+    echo "Download complete. Files saved to ${output_dir}/"
+    echo "Verification hashes saved to ${hash_file}"
+    echo ""
+    echo "Files downloaded:"
+    ls -lh "${output_dir}"
+    echo ""
+    echo "Verification hashes:"
+    cat "${hash_file}"
+    
+    return 0
+}
+
+# Validate command line arguments
+function validate_arguments() {
+    if [ $# -ne 1 ]; then
+        echo "Usage: $0 <size_in_bytes>"
```

Comment:
```suggestion
        echo "Usage: $0 <blob_size_in_bytes>"
```

Reply:
was unclear to me if the size was referring to G1 or G2 file. Explicitly saying its blob makes this more obvious what this file is doing imo

Reply:
Golang implementation has clearer names, so that it's obvious we are talking about blob size